apps/api/src/app/subscribers/e2e/mark-all-subscriber-messages.e2e.ts

```diff
@@ -0,0 +1,268 @@
+import { expect } from 'chai';
+import axios from 'axios';
+import { ChannelTypeEnum, MarkMessagesAsEnum } from '@novu/shared';
+import { UserSession } from '@novu/testing';
+import { NotificationTemplateEntity, MessageRepository } from '@novu/dal';
+
+const axiosInstance = axios.create();
+
+describe('Mark All Subscriber Messages - /subscribers/:subscriberId/messages/mark-all (POST)', function () {
+  let session: UserSession;
+  let template: NotificationTemplateEntity;
+  const messageRepository = new MessageRepository();
+
+  beforeEach(async () => {
+    session = new UserSession();
+    await session.initialize();
+    template = await session.createTemplate();
+    await messageRepository.deleteMany({
+      _environmentId: session.environment._id,
+      _subscriberId: session.subscriberId,
+    });
+  });
+
+  it("should throw not found when subscriberId doesn't exist", async function () {
+    const fakeSubscriberId = 'fake-subscriber-id';
+    try {
+      await markAllSubscriberMessagesAs(session, fakeSubscriberId, MarkMessagesAsEnum.READ);
```

Comment:
maybe it will be a good idea to throw an exception in the next line (or make any assertion) just to make sure that the flow got to the catch block as expected.